composer/core/state.py

```diff
@@ -511,7 +511,19 @@ def fsdp_enabled(self):
                 return True
         return False
 
-    def _get_state_metadata(self):
+    def _get_integrations_state_dict(self) -> Dict[str, Any]:
+        """Gets a dictionary of information about integrations to store in the state dict.
+
+        This metadata is used for loading things from state dict that need to be done outside
+        of the normal Composer load path (e.g. HuggingFace model/tokenizer).
+        """
+        from composer.models import HuggingFaceModel
+        integrations = {}
+        if isinstance(self.model, HuggingFaceModel):
+            integrations['huggingface'] = self.model.get_metadata()
+        return integrations
```

Comment:
can we try catch instead of type checking? more pythonic and lets users override if they really want to. If we do that, we can rename `get_metadata` to `get_huggingface_metadata`

Reply:
Hmm, I'm not sure that I want to use this code path to support users easily packing arbitrary info into the state dict. I'd prefer we have "control" over what goes into the `huggingface` key in the state dict, and provide a different code path for users storing whatever they want in it. I think that would be a different design question, and we would probably create an interface like `MetadataStoringModel`, and then if you implement that interface we store whatever metadata you want in the checkpoint under the key that you provide. Perhaps we do want to support something like that at some point, and the hugging face one could be refactored to the common interface at that point, but I'd probably wait for at least one more use case to pop up before that. Thoughts?

Reply:
FWIW, Daniel's suggestion makes sense to me.

Reply:
FWIW, Daniel's suggestion makes sense to me.

Reply:
Ok, I'm good with keeping it not user aligned. I would still suggest renaming to `get_huggingface_metadata` to be explicit. I also would prefer try catch over type checking because of the weird import. Iirc python is EAFP > LBYL. defer to you though

Reply:
Going to leave as is, I prefer explicit, and in this case the "except" would probably be more common than the "try" because most models are not `HuggingFaceModel`. Open to revisiting in the future if the imports in this function are getting unwieldy.